pandas/core/frame.py

```diff
@@ -9463,11 +9463,16 @@ def diff(self, periods: int = 1, axis: Axis = 0) -> DataFrame:
             periods = int(periods)
 
         axis = self._get_axis_number(axis)
+
         if axis == 1:
             if periods != 0:
                 # in the periods == 0 case, this is equivalent diff of 0 periods
                 #  along axis=0, and the Manager method may be somewhat more
                 #  performant, so we dispatch in that case.
+                if (self.dtypes == np.bool_).all():
+                    result = self ^ self.shift(periods=periods, axis=axis)
+                    result.iloc[:, :periods] = np.nan
```

Comment:
Hi @mikejigglypuff @jbrockmendel . I think this patch only works when periods>0, but periods could be negative.

Reply:
Sorry for the delay in responding, I've been sick for a few days. 
@yuanx749 As you said, the code was not able to handle when ``periods`` is negative, so I've committed a fixed version. Could you please take a look?